api/util/patron.py

```diff
@@ -0,0 +1,94 @@
+import datetime
+from api.config import Configuration
+from api.circulation_exceptions import *
+
+
+class PatronUtility(object):
```

Comment:
This code comes in from core basically unaltered except to become class methods of PatronUtility instead of instance methods of Patron.